test/BUILD.gn

```diff
@@ -416,6 +416,7 @@ test("brave_unit_tests") {
       "//components/gcm_driver/gcm_delayed_task_controller_unittest.cc",
       "//components/gcm_driver/gcm_driver_desktop_unittest.cc",
       "//components/gcm_driver/gcm_driver_unittest.cc",
+      "//components/password_manager/core/browser/ui/weak_check_utility_unittest.cc",
```

Comment:
It seems weird that we would have to add this file here, since it should be part of upstream's `components_unittest` target already. But it seems like we've had to do things like this before and it doesn't seem like we can build that upstream target at the moment anyway (and maybe it doesn't integrate with how we do our overrides anyway).

I think it'd be good to enter a follow-up issue to straighten out building that upstream target, so we can avoid this in the future.

Reply:
Some context on why I chose `brave_unit_tests`: originally, I wanted to keep this test in its original upstream target (`components_unittests`). When I tried building that target via `npm run build -- --target components_unittests`, I got this error:
```
../../components/page_load_metrics/browser/observers/ad_metrics/ads_page_load_metrics_observer_unittest.cc:915:21: error: field type 'FrameRemoteTester' is an abstract class
  915 |   FrameRemoteTester frame_remote_tester_;
      |                     ^
gen/third_party/blink/public/mojom/frame/frame.mojom.h:1217:16: note: unimplemented pure virtual method 'GetImageAt' in 'FrameRemoteTester'
 1217 |   virtual void GetImageAt(const ::gfx::Point& window_point, GetImageAtCallback callback) = 0;
```
This suggested that we don't run these tests at all — which I thought makes sense, as we'd be building an upstream target directly, while the `GetImageAt()` function is an addition of ours (`chromium_src/third_party/blink/public/mojom/frame/frame.mojom`):
```
[BraveExtend]
interface LocalFrame {
  GetImageAt(gfx.mojom.Point window_point)
      => (skia.mojom.BitmapN32? bitmap_image);
};
```
That said, I'm happy to get this test into `components_unittests` (where it would belong, ideally).

Reply:
Yeah, `component_unittests` is unfortunately broken. I've thought of fixing it (and `content_browsertests`), but I'm not sure I would like to face that challenge. There is the question however though if we can pull the just the target for the test itself, namely, `//components/password_manager/core/browser:unit_tests` as a dep of `brave_unit_tests`, but I'm that could also be broken for a variety of reasons.

Reply:
Quickly tried getting `npm run build -- --target components_unittests` to work. The compilation errors are easy to resolve, there are linkage-related ones as well. And then the tests themselves might be another round. If we want to enable these targets, we'll definitely want to give them a deeper look

Reply:
There's also the question if they pass.

Reply:
Yup, that's what I meant.

Reply:
https://github.com/brave/brave-browser/issues/42422